soccer/gameplay/tactics/positions/wing_defender.py

```diff
@@ -0,0 +1,78 @@
+import single_robot_behavior
+import behavior
+import robocup
+import math
+import main
+import skills
+import enum
+import constants
+import role_assignment
+import evaluation.defensive_positioning 
+
+
+class WingDefender(single_robot_behavior.SingleRobotBehavior):
+
+	class State(enum.Enum):
+		defending = 1
+		intercepting = 2
+
+	def __init__(self,
+				 goalside_ratio = .5,
+				 distance = .5,
+				 mark_robot = None,
+				 mark_point = None):
+		super().__init__(continuous=True)
+
+		self._goalside_ratio = goalside_ratio  	# Ratio of angle defending goal line versus attacker line
+		self._mark_robot = mark_robot 			# Robot we are defending against
+		self._distance = distance if distance >= 2*constants.Robot.Radius else 2*constants.Robot.Radius   # Distance from the point we are defending against
+
+		self._mark_pos = mark_point if mark_point != None \
+			else (mark_robot.pos if self.mark_robot != None else None)
+
+		self.kick_eval = robocup.KickEvaluator(main.system_state())
+
+		self.add_state(WingDefender.State.defending, 
+					   behavior.Behavior.State.running)
+
+		self.add_transition(behavior.Behavior.State.start,
+							WingDefender.State.defending, lambda: True,
+							'immediately')
+
+		# self.add_transition(
+		# 	WingDefender.State.defending, WingDefender.State.intercepting,
+		# 	lambda: self.ball_kicked,
+		# 	'ball kicked, moving to intercept')
+
+	def execute_defending(self):
+		self._mark_pos = self._mark_robot.pos if self._mark_robot != None else self.mark_pos
+		pt = self.calc_move_point()
+		move_skill = skills.move.Move(pt)
+		self.robot.move_to(pt)
+
+	def calc_move_point(self):
+		if self._mark_pos != None: 
+			goal_line, shot_pt = evaluation.defensive_positioning.goalside_mark_segment(self._mark_pos, self.robot, kick_eval=self.kick_eval)
```

Comment:
Throws an error :/ The next line will probably error too

Reply:
fixed, thanks